### PR TITLE
DROID-2561 Global search | Enhancement | Search with related objects

### PR DESCRIPTION
**Pull request by uburoiubu**

### CODE REVIEW

File: app/src/main/java/com/anytypeio/anytype/ui/search/GlobalSearchScreen.kt [hunk 1]
```diff
@@ -278,6 +384,54 @@ private fun GlobalSearchItem(
...
+                ) {
+                    DropdownMenuItem(
+                        onClick = { 
+                            onShowRelatedClicked(globalSearchItemView)
+                        }
+                    ) {
+                        Text(text = "Show related objects")
+                    }
+                }
+//                Dialog(
```
Comment (uburoiubu) [inline app/src/main/java/com/anytypeio/anytype/ui/search/GlobalSearchScreen.kt:409]: WIP, will be used in the next PR

File: presentation/src/main/java/com/anytypeio/anytype/presentation/search/GlobalSearchViewModel.kt [hunk 2]
```diff
@@ -130,6 +248,37 @@ class GlobalSearchViewModel(
             ) as T
         }
     }
+
+    private sealed class Mode {
```
Comment (konstantiniiv) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/search/GlobalSearchViewModel.kt:252]: add a description pls of what these modes are

File: presentation/src/main/java/com/anytypeio/anytype/presentation/search/GlobalSearchViewModel.kt [hunk 3]
```diff
@@ -55,47 +63,143 @@ class GlobalSearchViewModel(
...
+    )
+
+    private fun buildRelatedSearchFlow(
+        query: String,
+        mode: Mode.Related
+    ) = searchWithMeta
+        .stream(
+            Command.SearchWithMeta(
+                query = query,
+                limit = 50,
```
Comment (konstantiniiv) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/search/GlobalSearchViewModel.kt:105]: move 50 to const, pls

File: presentation/src/main/java/com/anytypeio/anytype/presentation/search/GlobalSearchViewModel.kt [hunk 4]
```diff
@@ -55,47 +63,143 @@ class GlobalSearchViewModel(
...
+                offset = 0,
+                keys = emptyList(),
+                filters = buildList {
+                    add(
```
Comment (konstantiniiv) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/search/GlobalSearchViewModel.kt:109]: it's better to have all filters inside ObjectSearchConstants class

**Review by konstantiniiv — APPROVED** [review]

File: presentation/src/main/java/com/anytypeio/anytype/presentation/search/GlobalSearchViewModel.kt (hunk 3 above)
Comment (uburoiubu) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/search/GlobalSearchViewModel.kt]: done

File: presentation/src/main/java/com/anytypeio/anytype/presentation/search/GlobalSearchViewModel.kt (hunk 2 above)
Comment (uburoiubu) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/search/GlobalSearchViewModel.kt:255]: done

File: presentation/src/main/java/com/anytypeio/anytype/presentation/search/GlobalSearchViewModel.kt (hunk 4 above)
Comment (uburoiubu) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/search/GlobalSearchViewModel.kt:114]: done

**Pull request merged**